**Linux** [Build 10505](https://ci.wildfly.org/viewLog.html?buildId=28303&buildTypeId=WFPR) outcome was **FAILURE** using a merge of 0d631241e6c8da1381d22a4a12ebb9edfae9cc5f
Summary: Tests failed: 6 (6 new), passed: 2276, ignored: 155 Build time: 00:55:58
### Failed tests
```
null: org.jboss.as.test.integration.deployment.SubDeploymentOperationsTestCase.testExplodeWarSubDeploymentArchiveDeployment: java.lang.AssertionError: Exploding subdeployment of archived deployment should fail, but outcome was {
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
	at org.jboss.as.test.integration.deployment.SubDeploymentOperationsTestCase.testExplodeWarSubDeploymentArchiveDeployment(SubDeploymentOperationsTestCase.java:170)
------- Stdout: -------
 [0m05:09:24,762 INFO  [org.jboss.as.repository] (management-handler-thread - 4) WFLYDR0001: Content added at location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/13/e4efedb134a7166240a0f4ec5055b287273743/content
 [0m [0m05:09:24,765 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-6) WFLYSRV0027: Starting deployment of "subdeployment-test.ear" (runtime-name: "subdeployment-test.ear")
 [0m [0m05:09:24,779 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-6) WFLYSRV0207: Starting subdeployment (runtime-name: "subdeployment-test-ejb.jar")
 [0m [0m05:09:24,782 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-5) WFLYSRV0207: Starting subdeployment (runtime-name: "subdeployment-test-web.war")
 [0m [0m05:09:24,799 INFO  [org.jboss.weld.deployer] (MSC service thread 1-5) WFLYWELD0003: Processing weld deployment subdeployment-test.ear
 [0m [0m05:09:24,812 INFO  [org.jboss.weld.deployer] (MSC service thread 1-6) WFLYWELD0003: Processing weld deployment subdeployment-test-web.war
 [0m [0m05:09:24,828 INFO  [org.jboss.weld.deployer] (MSC service thread 1-5) WFLYWELD0003: Processing weld deployment subdeployment-test-ejb.jar
 [0m [0m05:09:24,829 INFO  [org.jboss.as.ejb3.deployment] (MSC service thread 1-5) WFLYEJB0473: JNDI bindings for session bean named 'SimpleSLSB' in deployment unit 'subdeployment "subdeployment-test-ejb.jar" of deployment "subdeployment-test.ear"' are as follows:

	java:global/subdeployment-test/subdeployment-test-ejb/SimpleSLSB!org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.ejb.EJBBusinessInterface
	java:app/subdeployment-test-ejb/SimpleSLSB!org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.ejb.EJBBusinessInterface
	java:module/SimpleSLSB!org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.ejb.EJBBusinessInterface
	java:global/subdeployment-test/subdeployment-test-ejb/SimpleSLSB
	java:app/subdeployment-test-ejb/SimpleSLSB
	java:module/SimpleSLSB

 [0m [0m05:09:24,849 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 4) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m05:09:24,849 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 4) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m05:09:24,865 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 4) WFLYCLINF0002: Started client-mappings cache from ejb container
 [0m [33m05:09:24,886 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-3) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [33m05:09:24,892 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-3) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [33m05:09:24,895 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-3) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [0m05:09:24,974 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 11) WFLYUT0021: Registered web context: /subdeployment-test-web
 [0m [0m05:09:24,984 INFO  [org.jboss.as.server] (management-handler-thread - 4) WFLYSRV0010: Deployed "subdeployment-test.ear" (runtime-name : "subdeployment-test.ear")
 [0m [0m05:09:24,984 INFO  [org.jboss.as.controller] (management-handler-thread - 4) WFLYCTL0183: Service status report
WFLYCTL0185:    Newly corrected services:
      service jboss.module.service."deployment.deployment-two".main (no longer required)

 [0m [0m05:09:24,993 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 4) WFLYUT0022: Unregistered web context: /subdeployment-test-web
 [0m [0m05:09:25,012 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 25) WFLYCLINF0003: Stopped client-mappings cache from ejb container
 [0m [0m05:09:25,014 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-4) WFLYSRV0208: Stopped subdeployment (runtime-name: subdeployment-test-ejb.jar) in 24ms
 [0m [0m05:09:25,016 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-7) WFLYSRV0208: Stopped subdeployment (runtime-name: subdeployment-test-web.war) in 26ms
 [0m [0m05:09:25,019 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-7) WFLYSRV0028: Stopped deployment subdeployment-test.ear (runtime-name: subdeployment-test.ear) in 29ms
 [0m [0m05:09:25,025 INFO  [org.jboss.as.server] (management-handler-thread - 1) WFLYSRV0009: Undeployed "subdeployment-test.ear" (runtime-name: "subdeployment-test.ear")
 [0m [0m05:09:25,039 INFO  [org.jboss.as.repository] (management-handler-thread - 3) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/13/e4efedb134a7166240a0f4ec5055b287273743/content
 [0m [0m05:09:25,060 INFO  [org.jboss.as.repository] (management-handler-thread - 1) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/13/e4efedb134a7166240a0f4ec5055b287273743/content
 [0m


null: org.jboss.as.test.integration.deployment.SubDeploymentOperationsTestCase.testExplodeJarSubDeploymentArchiveDeployment: java.lang.AssertionError: Exploding subdeployment of archived deployment should fail, but outcome was {
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
	at org.jboss.as.test.integration.deployment.SubDeploymentOperationsTestCase.testExplodeJarSubDeploymentArchiveDeployment(SubDeploymentOperationsTestCase.java:156)
------- Stdout: -------
 [0m05:09:25,826 INFO  [org.jboss.as.repository] (management-handler-thread - 4) WFLYDR0001: Content added at location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/13/e4efedb134a7166240a0f4ec5055b287273743/content
 [0m [0m05:09:25,834 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-8) WFLYSRV0027: Starting deployment of "subdeployment-test.ear" (runtime-name: "subdeployment-test.ear")
 [0m [0m05:09:25,842 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-8) WFLYSRV0207: Starting subdeployment (runtime-name: "subdeployment-test-ejb.jar")
 [0m [0m05:09:25,842 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-8) WFLYSRV0207: Starting subdeployment (runtime-name: "subdeployment-test-web.war")
 [0m [0m05:09:25,853 INFO  [org.jboss.weld.deployer] (MSC service thread 1-3) WFLYWELD0003: Processing weld deployment subdeployment-test.ear
 [0m [0m05:09:25,861 INFO  [org.jboss.weld.deployer] (MSC service thread 1-1) WFLYWELD0003: Processing weld deployment subdeployment-test-web.war
 [0m [0m05:09:25,869 INFO  [org.jboss.weld.deployer] (MSC service thread 1-3) WFLYWELD0003: Processing weld deployment subdeployment-test-ejb.jar
 [0m [0m05:09:25,869 INFO  [org.jboss.as.ejb3.deployment] (MSC service thread 1-3) WFLYEJB0473: JNDI bindings for session bean named 'SimpleSLSB' in deployment unit 'subdeployment "subdeployment-test-ejb.jar" of deployment "subdeployment-test.ear"' are as follows:

	java:global/subdeployment-test/subdeployment-test-ejb/SimpleSLSB!org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.ejb.EJBBusinessInterface
	java:app/subdeployment-test-ejb/SimpleSLSB!org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.ejb.EJBBusinessInterface
	java:module/SimpleSLSB!org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.ejb.EJBBusinessInterface
	java:global/subdeployment-test/subdeployment-test-ejb/SimpleSLSB
	java:app/subdeployment-test-ejb/SimpleSLSB
	java:module/SimpleSLSB

 [0m [0m05:09:25,887 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 25) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m05:09:25,887 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 25) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m05:09:25,895 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 25) WFLYCLINF0002: Started client-mappings cache from ejb container
 [0m [33m05:09:25,905 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-3) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [33m05:09:25,905 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-3) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [33m05:09:25,906 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-3) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [0m05:09:25,955 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 11) WFLYUT0021: Registered web context: /subdeployment-test-web
 [0m [0m05:09:25,959 INFO  [org.jboss.as.server] (management-handler-thread - 4) WFLYSRV0010: Deployed "subdeployment-test.ear" (runtime-name : "subdeployment-test.ear")
 [0m [0m05:09:25,967 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 29) WFLYUT0022: Unregistered web context: /subdeployment-test-web
 [0m [0m05:09:25,978 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 8) WFLYCLINF0003: Stopped client-mappings cache from ejb container
 [0m [0m05:09:25,991 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-5) WFLYSRV0208: Stopped subdeployment (runtime-name: subdeployment-test-ejb.jar) in 28ms
 [0m [0m05:09:25,991 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-6) WFLYSRV0208: Stopped subdeployment (runtime-name: subdeployment-test-web.war) in 28ms
 [0m [0m05:09:25,994 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-6) WFLYSRV0028: Stopped deployment subdeployment-test.ear (runtime-name: subdeployment-test.ear) in 31ms
 [0m [0m05:09:25,999 INFO  [org.jboss.as.server] (management-handler-thread - 1) WFLYSRV0009: Undeployed "subdeployment-test.ear" (runtime-name: "subdeployment-test.ear")
 [0m [0m05:09:26,012 INFO  [org.jboss.as.repository] (management-handler-thread - 3) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/13/e4efedb134a7166240a0f4ec5055b287273743/content
 [0m [0m05:09:26,040 INFO  [org.jboss.as.repository] (management-handler-thread - 1) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/13/e4efedb134a7166240a0f4ec5055b287273743/content
 [0m


null: org.jboss.as.test.integration.batch.NoCdiTestCase.Failure when constructing test: java.lang.RuntimeException: Server state is 'reload-required' following test completion; tests must complete with the server in 'running' state
	at org.wildfly.core.testrunner.WildflyTestRunner.checkServerState(WildflyTestRunner.java:156)
	at org.wildfly.core.testrunner.WildflyTestRunner.runTearDownTasks(WildflyTestRunner.java:125)
	at org.wildfly.core.testrunner.WildflyTestRunner.run(WildflyTestRunner.java:101)


org.jboss.as.test.integration.ejb.timerservice.persistence.TimerServicePersistenceSecondTestCase.testPersistentCalendarTimer: java.lang.AssertionError: expected:<Hello> but was:<null>
	at org.jboss.as.test.integration.ejb.timerservice.persistence.TimerServicePersistenceSecondTestCase.testPersistentCalendarTimer(TimerServicePersistenceSecondTestCase.java:77)
------- Stdout: -------
	at org.jboss.as.test.integration.ejb.timerservice.persistence.TimerServicePersistenceSecondTestCase.testPersistentCalendarTimer(TimerServicePersistenceSecondTestCase.java:77)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.doRunTestMethod(ArquillianService.java:243)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.runTestMethod(ArquillianService.java:219)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:112)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:46)
	at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:237)
	at com.sun.jmx.mbeanserver.PerInterface.invoke(PerInterface.java:138)
	at com.sun.jmx.mbeanserver.MBeanSupport.invoke(MBeanSupport.java:252)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801)
	at org.jboss.as.jmx.PluggableMBeanServerImpl$TcclMBeanServer.invoke(PluggableMBeanServerImpl.java:1503)
	at org.jboss.as.jmx.PluggableMBeanServerImpl.invoke(PluggableMBeanServerImpl.java:724)
	at org.jboss.as.jmx.BlockingNotificationMBeanServer.invoke(BlockingNotificationMBeanServer.java:168)
	at org.jboss.remotingjmx.protocol.v2.ServerProxy$InvokeHandler.handle(ServerProxy.java:950)
	at org.jboss.remotingjmx.protocol.v2.ServerCommon$MessageReciever$1$1.run(ServerCommon.java:153)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor$1.run(ServerInterceptorFactory.java:75)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor$1.run(ServerInterceptorFactory.java:70)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.jboss.as.controller.AccessAuditContext.doAs(AccessAuditContext.java:149)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor.handleEvent(ServerInterceptorFactory.java:70)
	at org.jboss.remotingjmx.protocol.v2.ServerCommon$MessageReciever$1.run(ServerCommon.java:149)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

	at org.jboss.as.test.integration.ejb.timerservice.persistence.TimerServicePersistenceSecondTestCase.testPersistentCalendarTimer(TimerServicePersistenceSecondTestCase.java:77)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.doRunTestMethod(ArquillianService.java:243)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.runTestMethod(ArquillianService.java:219)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:112)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:46)
	at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:237)
	at com.sun.jmx.mbeanserver.PerInterface.invoke(PerInterface.java:138)
	at com.sun.jmx.mbeanserver.MBeanSupport.invoke(MBeanSupport.java:252)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801)
	at org.jboss.as.jmx.PluggableMBeanServerImpl$TcclMBeanServer.invoke(PluggableMBeanServerImpl.java:1503)
	at org.jboss.as.jmx.PluggableMBeanServerImpl.invoke(PluggableMBeanServerImpl.java:724)
	at org.jboss.as.jmx.BlockingNotificationMBeanServer.invoke(BlockingNotificationMBeanServer.java:168)
	at org.jboss.remotingjmx.protocol.v2.ServerProxy$InvokeHandler.handle(ServerProxy.java:950)
	at org.jboss.remotingjmx.protocol.v2.ServerCommon$MessageReciever$1$1.run(ServerCommon.java:153)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor$1.run(ServerInterceptorFactory.java:75)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor$1.run(ServerInterceptorFactory.java:70)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.jboss.as.controller.AccessAuditContext.doAs(AccessAuditContext.java:149)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor.handleEvent(ServerInterceptorFactory.java:70)
	at org.jboss.remotingjmx.protocol.v2.ServerCommon$MessageReciever$1.run(ServerCommon.java:149)


org.jboss.as.test.integration.ejb.timerservice.persistence.TimerServicePersistenceSecondTestCase.testTimerServiceCalled: java.lang.AssertionError: null
	at org.jboss.as.test.integration.ejb.timerservice.persistence.TimerServicePersistenceSecondTestCase.testTimerServiceCalled(TimerServicePersistenceSecondTestCase.java:56)
------- Stdout: -------
	at org.jboss.as.test.integration.ejb.timerservice.persistence.TimerServicePersistenceSecondTestCase.testTimerServiceCalled(TimerServicePersistenceSecondTestCase.java:56)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.doRunTestMethod(ArquillianService.java:243)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.runTestMethod(ArquillianService.java:219)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:112)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:46)
	at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:237)
	at com.sun.jmx.mbeanserver.PerInterface.invoke(PerInterface.java:138)
	at com.sun.jmx.mbeanserver.MBeanSupport.invoke(MBeanSupport.java:252)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801)
	at org.jboss.as.jmx.PluggableMBeanServerImpl$TcclMBeanServer.invoke(PluggableMBeanServerImpl.java:1503)
	at org.jboss.as.jmx.PluggableMBeanServerImpl.invoke(PluggableMBeanServerImpl.java:724)
	at org.jboss.as.jmx.BlockingNotificationMBeanServer.invoke(BlockingNotificationMBeanServer.java:168)
	at org.jboss.remotingjmx.protocol.v2.ServerProxy$InvokeHandler.handle(ServerProxy.java:950)
	at org.jboss.remotingjmx.protocol.v2.ServerCommon$MessageReciever$1$1.run(ServerCommon.java:153)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor$1.run(ServerInterceptorFactory.java:75)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor$1.run(ServerInterceptorFactory.java:70)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.jboss.as.controller.AccessAuditContext.doAs(AccessAuditContext.java:149)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor.handleEvent(ServerInterceptorFactory.java:70)
	at org.jboss.remotingjmx.protocol.v2.ServerCommon$MessageReciever$1.run(ServerCommon.java:149)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

	at org.jboss.as.test.integration.ejb.timerservice.persistence.TimerServicePersistenceSecondTestCase.testTimerServiceCalled(TimerServicePersistenceSecondTestCase.java:56)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.doRunTestMethod(ArquillianService.java:243)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.runTestMethod(ArquillianService.java:219)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:112)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:46)
	at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:237)
	at com.sun.jmx.mbeanserver.PerInterface.invoke(PerInterface.java:138)
	at com.sun.jmx.mbeanserver.MBeanSupport.invoke(MBeanSupport.java:252)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801)
	at org.jboss.as.jmx.PluggableMBeanServerImpl$TcclMBeanServer.invoke(PluggableMBeanServerImpl.java:1503)
	at org.jboss.as.jmx.PluggableMBeanServerImpl.invoke(PluggableMBeanServerImpl.java:724)
	at org.jboss.as.jmx.BlockingNotificationMBeanServer.invoke(BlockingNotificationMBeanServer.java:168)
	at org.jboss.remotingjmx.protocol.v2.ServerProxy$InvokeHandler.handle(ServerProxy.java:950)
	at org.jboss.remotingjmx.protocol.v2.ServerCommon$MessageReciever$1$1.run(ServerCommon.java:153)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor$1.run(ServerInterceptorFactory.java:75)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor$1.run(ServerInterceptorFactory.java:70)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.jboss.as.controller.AccessAuditContext.doAs(AccessAuditContext.java:149)
	at org.jboss.as.jmx.ServerInterceptorFactory$Interceptor.handleEvent(ServerInterceptorFactory.java:70)
	at org.jboss.remotingjmx.protocol.v2.ServerCommon$MessageReciever$1.run(ServerCommon.java:149)


org.jboss.as.test.integration.ejb.timerservice.serialization.TimerServiceSerializationSecondTestCase.testTimerServiceRestoredWithCorrectInfo: java.lang.AssertionError: null
	at org.jboss.as.test.integration.ejb.timerservice.serialization.TimerServiceSerializationSecondTestCase.testTimerServiceRestoredWithCorrectInfo(TimerServiceSerializationSecondTestCase.java:55)
------- Stdout: -------
 [0m05:10:42,040 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 73) WFLYUT0022: Unregistered web context: /testSchedule
 [0m [0m05:10:42,083 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-1) WFLYSRV0028: Stopped deployment testSchedule.war (runtime-name: testSchedule.war) in 44ms
 [0m [0m05:10:42,087 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 73) WFLYCLINF0003: Stopped client-mappings cache from ejb container
 [0m [0m05:10:42,095 INFO  [org.jboss.as.repository] (management-handler-thread - 8) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/c7/3250fd60f749a66db5face7daf70364c7e5649/content
 [0m [0m05:10:42,095 INFO  [org.jboss.as.server] (management-handler-thread - 8) WFLYSRV0009: Undeployed "testSchedule.war" (runtime-name: "testSchedule.war")
 [0m [0m05:10:42,133 INFO  [org.jboss.as.repository] (management-handler-thread - 5) WFLYDR0001: Content added at location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/c6/6c7e4a62398e2f58ccc1fe4b1eecea9d2ed770/content
 [0m [0m05:10:42,136 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-1) WFLYSRV0027: Starting deployment of "testTimerServiceSerialization.ear" (runtime-name: "testTimerServiceSerialization.ear")
 [0m [0m05:10:42,148 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-4) WFLYSRV0207: Starting subdeployment (runtime-name: "b.jar")
 [0m [0m05:10:42,149 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-7) WFLYSRV0207: Starting subdeployment (runtime-name: "a.jar")
 [0m [0m05:10:42,148 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0207: Starting subdeployment (runtime-name: "c.jar")
 [0m [33m05:10:42,173 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-6) WFLYSRV0018: Deployment "deployment.testTimerServiceSerialization.ear.b.jar" is using a private module ("org.wildfly.security.manager:main") which may be changed or removed in future versions without notice.
 [0m [33m05:10:42,175 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-7) WFLYSRV0018: Deployment "deployment.testTimerServiceSerialization.ear" is using a private module ("org.wildfly.security.manager:main") which may be changed or removed in future versions without notice.
 [0m [33m05:10:42,176 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-5) WFLYSRV0018: Deployment "deployment.testTimerServiceSerialization.ear.a.jar" is using a private module ("org.wildfly.security.manager:main") which may be changed or removed in future versions without notice.
 [0m [33m05:10:42,178 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-2) WFLYSRV0018: Deployment "deployment.testTimerServiceSerialization.ear.c.jar" is using a private module ("org.wildfly.security.manager:main") which may be changed or removed in future versions without notice.
 [0m [0m05:10:42,178 INFO  [org.jboss.weld.deployer] (MSC service thread 1-2) WFLYWELD0003: Processing weld deployment testTimerServiceSerialization.ear
 [0m [0m05:10:42,186 INFO  [org.jboss.weld.deployer] (MSC service thread 1-5) WFLYWELD0003: Processing weld deployment b.jar
 [0m [0m05:10:42,186 INFO  [org.jboss.weld.deployer] (MSC service thread 1-7) WFLYWELD0003: Processing weld deployment c.jar
 [0m [0m05:10:42,193 INFO  [org.jboss.weld.deployer] (MSC service thread 1-2) WFLYWELD0003: Processing weld deployment a.jar
 [0m [0m05:10:42,194 INFO  [org.jboss.as.ejb3.deployment] (MSC service thread 1-2) WFLYEJB0473: JNDI bindings for session bean named 'TimerServiceSerializationBean' in deployment unit 'subdeployment "a.jar" of deployment "testTimerServiceSerialization.ear"' are as follows:

	java:global/testTimerServiceSerialization/a/TimerServiceSerializationBean!org.jboss.as.test.integration.ejb.timerservice.serialization.TimerServiceSerializationBean
	java:app/a/TimerServiceSerializationBean!org.jboss.as.test.integration.ejb.timerservice.serialization.TimerServiceSerializationBean
	java:module/TimerServiceSerializationBean!org.jboss.as.test.integration.ejb.timerservice.serialization.TimerServiceSerializationBean
	java:global/testTimerServiceSerialization/a/TimerServiceSerializationBean
	java:app/a/TimerServiceSerializationBean
	java:module/TimerServiceSerializationBean

 [0m [0m05:10:42,215 INFO  [org.jboss.as.arquillian] (MSC service thread 1-6) Arquillian deployment detected: ArquillianConfig[service=jboss.arquillian.config."a.jar",unit=a.jar,tests=[org.jboss.as.test.integration.ejb.timerservice.serialization.TimerServiceSerializationSecondTestCase]]
 [0m [0m05:10:42,228 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 73) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m05:10:42,228 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 73) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m05:10:42,232 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 73) WFLYCLINF0002: Started client-mappings cache from ejb container
 [0m [33m05:10:42,256 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-2) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [33m05:10:42,257 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-2) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [33m05:10:42,258 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-2) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [33m05:10:42,258 WARN  [org.jboss.weld.Bootstrap] (MSC service thread 1-2) WELD-000169: Jandex cannot distinguish inner and static nested classes! Update Jandex to 2.0.3.Final version or newer to improve scanning performance.
 [0m [0m05:10:42,393 INFO  [org.jboss.as.server] (management-handler-thread - 5) WFLYSRV0010: Deployed "testTimerServiceSerialization.ear" (runtime-name : "testTimerServiceSerialization.ear")
	at org.jboss.as.test.integration.ejb.timerservice.serialization.TimerServiceSerializationSecondTestCase.testTimerServiceRestoredWithCorrectInfo(TimerServiceSerializationSecondTestCase.java:55)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.doRunTestMethod(ArquillianService.java:243)
	at org.jboss.as.arquillian.service.ArquillianService$ExtendedJMXTestRunner.runTestMethod(ArquillianService.java:219)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:112)
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:46)
	at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:237)
	at com.sun.jmx.mbeanserver.PerInterface.invoke(PerInterface.java:138)
	at com.sun.jmx.mbeanserver.MBeanSupport.invoke(MBeanSupport.java:252)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801)
	at org.jboss.as.jmx.PluggableMBeanServerImpl$TcclMBeanServer.invoke(PluggableMBeanServerImpl.java:1503)
	at org.jboss.as.jmx.PluggableMBeanServerImpl.invoke(PluggableMBeanServerImpl.java:724)


```

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/SubDeploymentOperationsTestCase.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/SubDeploymentOperationsTestCase.java
@@ -1,0 +1,262 @@
+/*
+ * Copyright (C) 2016 Red Hat, inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301  USA
+ */
+
+package org.jboss.as.test.integration.deployment;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.client.Operation;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.ejb.EJBBusinessInterface;
+import org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.ejb.SimpleSLSB;
+import org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.servlet.EjbInvokingServlet;
+import org.jboss.as.test.integration.deployment.classloading.ear.subdeployments.servlet.HelloWorldServlet;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.dmr.ModelNode;
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tests to check that the subdeployments function properly with other subdeployments after using
+ * the explode operation.
+ * @author <a href="mailto:mjurc@redhat.com">Michal Jurc</a> (c) 2016 Red Hat, Inc.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class SubDeploymentOperationsTestCase {
+
+    private static final Logger logger = Logger.getLogger(SubDeploymentOperationsTestCase.class);
+
+    private static final String TEST_DEPLOYMENT_NAME = "subdeployment-test.ear";
+    private static final String JAR_SUBDEPLOYMENT_NAME = "subdeployment-test-ejb.jar";
+    private static final String WAR_SUBDEPLOYMENT_NAME = "subdeployment-test-web.war";
+    private static final String ARCHIVED_DEPLOYMENT_ERROR_CODE = "WFLYSRV0258";
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    @Before
+    public void setUpDeployment() throws Exception {
+        ModelNode op = new ModelNode();
+        op.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.READ_RESOURCE_OPERATION);
+        op.get(ModelDescriptionConstants.OP_ADDR).set(ModelDescriptionConstants.DEPLOYMENT, TEST_DEPLOYMENT_NAME);
+        ModelNode result = managementClient.getControllerClient().execute(op);
+        if (Operations.isSuccessfulOutcome(result)) {
+            undeploy(TEST_DEPLOYMENT_NAME);
+            remove(TEST_DEPLOYMENT_NAME);
+        }
+
+        result = initialDeploy();
+        Assert.assertTrue("Failure to set the initial development up: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+
+        result = undeploy(TEST_DEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to undeploy the initial development: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+    }
+
+    @After
+    public void cleanUpDeployment() throws Exception {
+        ModelNode op = new ModelNode();
+        op.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.READ_RESOURCE_OPERATION);
+        op.get(ModelDescriptionConstants.OP_ADDR).set(ModelDescriptionConstants.DEPLOYMENT, TEST_DEPLOYMENT_NAME);
+        ModelNode result = managementClient.getControllerClient().execute(op);
+        if (Operations.isSuccessfulOutcome(result)) {
+            undeploy(TEST_DEPLOYMENT_NAME);
+            remove(TEST_DEPLOYMENT_NAME);
+        }
+    }
+
+    @Test
+    public void testExplodeJarSubDeployment() throws Exception {
+        ModelNode result = explode(TEST_DEPLOYMENT_NAME, "");
+        Assert.assertTrue("Failure to explode the initial deployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        result = explode(TEST_DEPLOYMENT_NAME, JAR_SUBDEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to explode JAR subdeployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        result = deploy(TEST_DEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to redeploy the deployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        testEjbClassAvailableInServlet();
+    }
+
+    @Test
+    public void testExplodeWarSubDeployment() throws Exception {
+        ModelNode result = explode(TEST_DEPLOYMENT_NAME, "");
+        Assert.assertTrue("Failure to explode the initial deployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        result = explode(TEST_DEPLOYMENT_NAME, WAR_SUBDEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to explode WAR subdeployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        result = deploy(TEST_DEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to redeploy the deployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        testEjbClassAvailableInServlet();
+    }
+
+    @Test
+    public void testExplodeJarAndWarSubDeployment() throws Exception {
+        ModelNode result = explode(TEST_DEPLOYMENT_NAME, "");
+        Assert.assertTrue("Failure to explode the initial deployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        result = explode(TEST_DEPLOYMENT_NAME, JAR_SUBDEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to explode JAR subdeployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        result = explode(TEST_DEPLOYMENT_NAME, WAR_SUBDEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to explode WAR subdeployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        result = deploy(TEST_DEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to redeploy the deployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        testEjbClassAvailableInServlet();
+    }
+
+    @Test
+    public void testExplodeJarSubDeploymentArchiveDeployment() throws Exception {
+        ModelNode result = explode(TEST_DEPLOYMENT_NAME, JAR_SUBDEPLOYMENT_NAME);
+        Assert.assertFalse("Exploding JAR subdeployment of archived deployment should fail, but outcome was " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        String failure = Operations.getFailureDescription(result).asString();
+        Assert.assertTrue("Exploding JAR subdeployment of archived deployment failed with wrong reason: " + failure,
+                failure.contains(ARCHIVED_DEPLOYMENT_ERROR_CODE));
+        result = deploy(TEST_DEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to redeploy the deployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        testEjbClassAvailableInServlet();
+    }
+
+    @Test
+    public void testExplodeWarSubDeploymentArchiveDeployment() throws Exception {
+        ModelNode result = explode(TEST_DEPLOYMENT_NAME, WAR_SUBDEPLOYMENT_NAME);
+        Assert.assertFalse("Exploding WAR subdeployment of archived deployment should fail, but outcome was " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        String failure = Operations.getFailureDescription(result).asString();
+        Assert.assertTrue("Exploding WAR subdeployment of archived deployment failed with wrong reason: " + failure,
+                failure.contains(ARCHIVED_DEPLOYMENT_ERROR_CODE));
+        result = deploy(TEST_DEPLOYMENT_NAME);
+        Assert.assertTrue("Failure to redeploy the deployment: " + result.toString(),
+                Operations.isSuccessfulOutcome(result));
+        testEjbClassAvailableInServlet();
+    }
+
+    private ModelNode initialDeploy() throws Exception {
+        ModelNode result;
+        List<InputStream> attachments = new ArrayList<>();
+
+        JavaArchive ejbJar = ShrinkWrap.create(JavaArchive.class, JAR_SUBDEPLOYMENT_NAME)
+                .addClasses(EJBBusinessInterface.class, SimpleSLSB.class);
+        WebArchive war = ShrinkWrap.create(WebArchive.class, WAR_SUBDEPLOYMENT_NAME)
+                .addClasses(HelloWorldServlet.class, EjbInvokingServlet.class);
+        EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, TEST_DEPLOYMENT_NAME)
+                .addAsModule(ejbJar)
+                .addAsModule(war);
+
+        try (InputStream is = ear.as(ZipExporter.class).exportAsInputStream()) {
+            ModelNode compositeOp = new ModelNode();
+            ModelNode addOp = new ModelNode();
+            addOp.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.ADD);
+            addOp.get(ModelDescriptionConstants.OP_ADDR).set(ModelDescriptionConstants.DEPLOYMENT, TEST_DEPLOYMENT_NAME);
+            addOp.get(ModelDescriptionConstants.CONTENT).add(ModelDescriptionConstants.INPUT_STREAM_INDEX, 0);
+            ModelNode content = new ModelNode();
+            content.get(ModelDescriptionConstants.INPUT_STREAM_INDEX).set(0);
+            ModelNode deployOp = new ModelNode();
+            deployOp.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.DEPLOY);
+            deployOp.get(ModelDescriptionConstants.OP_ADDR).set(ModelDescriptionConstants.DEPLOYMENT, TEST_DEPLOYMENT_NAME);
+            compositeOp.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.COMPOSITE);
+            compositeOp.get(ModelDescriptionConstants.STEPS).setEmptyList();
+            compositeOp.get(ModelDescriptionConstants.STEPS).add(addOp);
+            compositeOp.get(ModelDescriptionConstants.STEPS).add(deployOp);
+
+            attachments.add(is);
+
+            result = managementClient.getControllerClient().execute(Operation.Factory.create(compositeOp, attachments));
+        }
+
+        return result;
+    }
+
+    private ModelNode deploy(String deployment) throws Exception {
+        ModelNode deployOp = new ModelNode();
+        deployOp.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.DEPLOY);
+        deployOp.get(ModelDescriptionConstants.OP_ADDR).set(ModelDescriptionConstants.DEPLOYMENT, deployment);
+        return managementClient.getControllerClient().execute(deployOp);
+    }
+
+    private ModelNode explode(String deployment, String path) throws Exception {
+        ModelNode explodeOp = new ModelNode();
+        explodeOp.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.EXPLODE);
+        explodeOp.get(ModelDescriptionConstants.OP_ADDR).set(ModelDescriptionConstants.DEPLOYMENT, deployment);
+        if (!path.isEmpty()) {
+            explodeOp.get(ModelDescriptionConstants.PATH).set(path);
+        }
+        return managementClient.getControllerClient().execute(explodeOp);
+    }
+
+    private ModelNode undeploy(String deployment) throws Exception {
+        ModelNode undeployOp = new ModelNode();
+        undeployOp.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.UNDEPLOY);
+        undeployOp.get(ModelDescriptionConstants.OP_ADDR).set(ModelDescriptionConstants.DEPLOYMENT, deployment);
+        return managementClient.getControllerClient().execute(undeployOp);
+    }
+
+    private ModelNode remove(String deployment) throws Exception {
+        ModelNode removeOp = new ModelNode();
+        removeOp.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.REMOVE);
+        removeOp.get(ModelDescriptionConstants.OP_ADDR).set(ModelDescriptionConstants.DEPLOYMENT, deployment);
+        return managementClient.getControllerClient().execute(removeOp);
+    }
+
+    private void testEjbClassAvailableInServlet() throws Exception {
+        final HttpClient httpClient = HttpClients.createDefault();
+        final String message = "JBossEAP";
+        final String requestURL = TestSuiteEnvironment.getHttpUrl().toString() + "/subdeployment-test-web" + HelloWorldServlet.URL_PATTERN + "?" + HelloWorldServlet.PARAMETER_NAME + "=" + message;
+        final HttpGet request = new HttpGet(requestURL);
+        final HttpResponse response = httpClient.execute(request);
+        final HttpEntity entity = response.getEntity();
+        Assert.assertNotNull("Response message from servlet was null", entity);
+        final String responseMessage = EntityUtils.toString(entity);
+        Assert.assertEquals("Unexpected echo message from servlet at " + requestURL, message, responseMessage);
+    }
+
+}

